### PR TITLE
Dispose KeepAliveHandle. [prod-leak-fix]

### PR DESCRIPTION
**Pull request by polina-c**
The issue was detected by [leak-tracker](https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/memory/panes/leaks/LEAK_TRACKING.md) for Gallery.

We will test cover the change after the leak tracker release.

### CODE REVIEW

**Comment by polina-c** [comment]
Requested exemption: https://discord.com/channels/608014603317936148/608018585025118217/1001186592179896450

**Comment by Hixie** [comment]
test-exempt: will be tested by https://github.com/flutter/flutter/issues/108322

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/automatic_keep_alive.dart [hunk 1]
```diff
@@ -354,6 +354,7 @@ mixin AutomaticKeepAliveClientMixin<T extends StatefulWidget> on State<T> {
 
   void _releaseKeepAlive() {
     _keepAliveHandle!.release();
+    _keepAliveHandle!.dispose();
```
Comment (dnfield) [inline packages/flutter/lib/src/widgets/automatic_keep_alive.dart:357]: consider adding a comment that dispose and release do not imply each other. Alternatively, consider renaming release to something like notify.

**Review by dnfield — APPROVED** [review]

File: packages/flutter/lib/src/widgets/automatic_keep_alive.dart (hunk 1 above)
Comment (polina-c) [reply packages/flutter/lib/src/widgets/automatic_keep_alive.dart:358]: It seems renaming the method 'release' will introduce breaking change. Added comment.

**Pull request merged**